src/components/shared/globalTypes.ts

```diff
@@ -153,31 +156,44 @@ export class Directory extends FileSystemObject {
   changeCurrentWorkingDirectory(
     currentWorkingDirectory: Directory,
     path: string
-  ): Directory | undefined {
+  ): Directory | string {
     currentWorkingDirectory.isCurrentDirectory = false;
     let newCwd: Directory | File | undefined;
 
-    if (!path.startsWith('/')) {
-      newCwd = currentWorkingDirectory.getFileSystemObjectFromPath(path);
-    } else {
+    // If the path is absolute, we start from the root directory and find the new cwd
+    if (path.startsWith('/')) {
       newCwd = this.getFileSystemObjectFromPath(path);
+    } else {
+      newCwd = currentWorkingDirectory.getFileSystemObjectFromPath(path);
     }
 
-    if (!newCwd || !newCwd.isDirectory) return undefined;
+    // If the new cwd is a file or does not exist, we can't cd into it
+    if (!newCwd) {
+      return `cd: '${path}': No such directory`;
+    }
+    if (newCwd instanceof File) {
+      return `cd: '${path}': Not a directory`;
+    }
     (newCwd as Directory).isCurrentDirectory = true;
     return newCwd as Directory;
   }
 
   getFileSystemObjectFromPath(path: string): Directory | File | undefined {
     if (path === this.path) return this;
 
+    // Removes the trailing slash if it exists
     if (path.charAt(path.length - 1) == '/') {
       path = path.slice(0, -1);
     }
 
-    const children = path.includes('/') ? path.split('/').slice(1) : [path];
+    const children = path.includes('/')
+      ? path.split('/').slice(path.startsWith('/') ? 1 : 0)
+      : [path];
+
     let currentDirectory = <Directory>this;
     let currentFsObject: Directory | File | undefined;
+
+    // Traverse the path to find the new cwd and return the fs object
     for (let i = 0; i < children.length; i++) {
```

Comment:
Replace it with this for...of loop - `for (const child of children)`

Reply:
Agreed, index is not used in this loop I think.